th2_data_services/interfaces/events_tree/events_tree_collection.py

```diff
@@ -46,12 +57,19 @@ def __init__(
             preserve_body: If True it will preserve 'body' field in the Events.
             stub: If True it will create stub when event is broken.
         """
+        self._id = id(self)
         self._preserve_body = preserve_body
         self._roots: List[EventsTree] = []
         self._parentless: Optional[List[EventsTree]] = None
         self._detached_nodes: Dict[Optional[str], List[Node]] = defaultdict(list)  # {parent_event_id: [Node1, ..]}
         self._stub_status = stub
         self._data_source = data_source
+        self._logger = _EventsTreeCollectionLogger(logger, {"id": self._id})
+        self._warnings = []
+
+        # It used to indicate the number of current iteration of the ETC object.
+        # It's required if the same instance iterates several times in for-in loops.
+        # Similar code in the Data object
```

Comment:
What code is this comment for?

Reply:
Will fixed in next commit.

Reply:
Removed from events_tree_collection.py https://github.com/th2-net/th2-data-services/pull/131/commits/0460ed688d47f24a298ed465e68c81b9cfbf7118